packages/@ember/-internals/glimmer/lib/resolver.ts

```diff
@@ -166,10 +166,12 @@ export default class ResolverImpl implements RuntimeResolver<Owner>, CompileTime
   lookupHelper(name: string, owner: Owner): Option<HelperDefinitionState> {
     assert(
       `You attempted to overwrite the built-in helper "${name}" which is not allowed. Please rename the helper.`,
-      !(BUILTIN_HELPERS[name] && owner.hasRegistration(`helper:${name}`))
+      !(
+        (BUILTIN_HELPERS as Record<string, object>)[name] && owner.hasRegistration(`helper:${name}`)
```

Comment:
If I'm reading the code right above, this is now how `BUILTIN_HELPERS` is defined type-wise, so the cast should be unnecessary? (And the same throughout below, of course.)